File: benchmark/output/python-pygame/gemini/multi_wave_spawner.py
```python
import pygame
import sys
import random
import math

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Wave Spawner")
clock = pygame.time.Clock()

# Player state
px, py = 400.0, 300.0
player_speed = 250.0

# Enemy parallel lists
ex = []
ey = []
evx = []
evy = []
elife = []

# Wave state
wave = 1
spawn_timer = 2.0
enemies_per_wave = 3
game_over = False

font = pygame.font.Font(None, 36)

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    if not game_over:
        keys = pygame.key.get_pressed()
        if keys[pygame.K_LEFT]:
            px -= player_speed * dt
        if keys[pygame.K_RIGHT]:
            px += player_speed * dt
        if keys[pygame.K_UP]:
            py -= player_speed * dt
        if keys[pygame.K_DOWN]:
            py += player_speed * dt

        spawn_timer -= dt
        if spawn_timer <= 0.0:
            for _ in range(enemies_per_wave):
                edge = random.uniform(0.0, 4.0)
                if edge < 1.0:
                    nx = random.uniform(0.0, 800.0)
                    ny = 0.0
                elif edge < 2.0:
                    nx = 800.0
                    ny = random.uniform(0.0, 600.0)
                elif edge < 3.0:
                    nx = random.uniform(0.0, 800.0)
                    ny = 600.0
                else:
                    nx = 0.0
                    ny = random.uniform(0.0, 600.0)

                dx = px - nx
                dy = py - ny
                dist = math.hypot(dx, dy)
                if dist != 0:
                    dx /= dist
                    dy /= dist
                
                speed = 80.0 + wave * 20.0
                
                ex.append(nx)
                ey.append(ny)
                evx.append(dx * speed)
                evy.append(dy * speed)
                elife.append(1.0)

            wave += 1
            enemies_per_wave = 3 + wave
            spawn_timer = max(0.5, 3.0 - wave * 0.2)

        for i in range(len(ex)):
            if elife[i] > 0:
                ex[i] += evx[i] * dt
                ey[i] += evy[i] * dt
                elife[i] -= dt

                dist = math.hypot(px - ex[i], py - ey[i])
                if dist < 20.0:  # 12 (player radius) + 8 (half of 16x16 rect)
                    game_over = True

    screen.fill((0, 0, 0))

    pygame.draw.circle(screen, (255, 255, 255), (int(px), int(py)), 12)

    for i in range(len(ex)):
        if elife[i] > 0:
            pygame.draw.rect(screen, (255, 255, 255), (int(ex[i]

File: vibe-lang/benchmark/output/lua-love/openai/multi_wave_spawner.lua
local playerX = 400
local playerY = 300
local playerSpeed = 250
local playerRadius = 12

local wave = 1
local spawn_timer = 2.0
local enemies_per_wave = 3

local ex = {}
local ey = {}
local evx = {}
local evy = {}
local elife = {}

local game_over = false

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Wave-Based Enemy Spawner")
end

function love.update(dt)
    if not game_over then
        -- Player movement
        if love.keyboard.isDown("left") then
            playerX = playerX - playerSpeed * dt
        end
        if love.keyboard.isDown("right") then
            playerX = playerX + playerSpeed * dt
        end
        if love.keyboard.isDown("up") then
            playerY = playerY - playerSpeed * dt
        end
        if love.keyboard.isDown("down") then
            playerY = playerY + playerSpeed * dt
        end

        -- Spawn enemies
        spawn_timer = spawn_timer - dt
        if spawn_timer <= 0 then
            for i = 1, enemies_per_wave do
                local edge = love.math.random() * 4
                local x, y
                if edge < 1 then
                    x = love.math.random() * 800
                    y = 0
                elseif edge < 2 then
                    x = 800
                    y = love.math.random() * 600
                elseif edge < 3 then
                    x = love.math.random() * 800
                    y = 600
                else
                    x = 0
                    y = love.math.random() * 600
                end

                local dx = playerX - x
                local dy = playerY - y
                local dist = math.sqrt(dx * dx + dy * dy)
                local speed = 80 + wave * 20
                local vx = (dx / dist) * speed
                local vy = (dy / dist) * speed

                table.insert(ex, x)
                table.insert(ey, y)
                table.insert(evx, vx)
                table.insert(evy, vy)
                table.insert(elife, 1.0)
            end

            wave = wave + 1
            enemies_per_wave = 3 + wave
            spawn_timer = math.max(0.5, 3.0 - wave * 0.2)
        end

        -- Update enemies
        for i = #ex, 1, -1 do
            ex[i] = ex[i] + evx[i] * dt
            ey[i] = ey[i] + evy[i] * dt
            elife[i] = elife[i] - dt

            -- Check collision with player
            local dx = playerX - ex[i]
            local dy = playerY - ey[i]
            local distance = math.sqrt(dx * dx + dy * dy)
            if distance < playerRadius + 8 then
                game_over = true
            end

            -- Remove dead enemies
            if elife[i] <= 0 then
                table.remove(ex, i)
                table.remove(ey, i)
                table.remove(evx, i)
                table.remove(evy, i)
                table.remove(elife, i)
            end
        end
    end
end

function love.draw()
    -- Draw player
    love.graphics.setColor(1, 1, 1)
    love.graphics.circle("fill", playerX, playerY, playerRadius)

    -- Draw enemies
    for i = 1, #ex do
        if elife[i] > 0 then
            love.graphics.rectangle("fill", ex[i] - 8, ey[i] - 8, 16, 16)
        end
    end

    -- Draw wave number
    love.graphics.print("Wave: " .. wave, 10, 10)

    -- Draw game over
    if game_over then
        love.graphics.printf("GAME OVER", 0, love.graphics.getHeight() / 2 - 10, love.graphics.getWidth(), "center")
    end
end